### DISCUSSION TITLE
Add a generic to useMultiStyleConfig theme helper

### DISCUSSION THREAD

**Discussion by isBatak** [chakra-ui/chakra-ui#8271] [Ideas]
### Description

Developer should be able do define multi-style parts types for the custom components.

For example:
```ts
export type AccordionMultiStyleConfig = Record<typeof accordionAnatomy.__type, SystemStyleObject>;

const styles = useMultiStyleConfig('Accordion', props) as AccordionMultiStyleConfig;

<chakra.div __css={styles.container} />
```

`styles.container` is autocompleted.

### Problem Statement/Justification

Right now we have to assert the type.
```ts
const styles = useMultiStyleConfig('Accordion', props) as AccordionMultiStyleConfig;
``` 
This is not safe.

### Proposed Solution or API

Add support for generic like this:
```ts
const styles = useMultiStyleConfig<AccordionMultiStyleConfig>('Accordion', props);
``` 
or create a type with all of the multi-style types and use it internally.
```ts
type MultiStyleConfig = {
 'Accordion':  typeof accordionAnatomy.__type;
}

export function useMultiStyleConfig<ThemeKey extends keyof MultiStyleConfig>(
  themeKey: ThemeKey,
) {
  return {} as Record<
    MultiStyleConfig[ThemeKey],
    string
  >
}
```

**Comment by norbert-kulus-blockydevs** [comment]
I really like the approach. Are there any updates on it?

**Reply by elboletaire** [reply]
Did you upvote the discussion? I suspect chakra developers decided to only take care of the more voted discussions, since they moved all issues from the issues section to the discussions one (without any kind of advice or warning btw), and losing any reactions they had.

**Comment by segunadebayo** [comment]
`useMultiStyleConfig` is a v2 helper. v3 uses recipes.
